src/netex_idf/offers.rs

```diff
@@ -314,6 +370,73 @@ where
         .collect()
 }
 
+type RoutingConstraintZone = Vec<Idx<StopPoint>>;
+fn parse_routing_constraint_zones<'a, I>(
+    rcz_elements: I,
+    map_schedule_stop_point_quay: &HashMap<String, String>,
+    collections: &Collections,
+) -> Vec<RoutingConstraintZone>
+where
+    I: Iterator<Item = &'a Element>,
+{
+    rcz_elements
+        .filter_map(|rcz_element| {
+            let mut stop_point_idxs = RoutingConstraintZone::new();
+            for scheduled_stop_point_ref_element in rcz_element
+                .only_child("members")
+                .iter()
+                .flat_map(|members| members.children())
+            {
+                if let Some(stop_point_idx) = scheduled_stop_point_ref_element
+                    .attribute::<String>("ref")
+                    .and_then(|ssp_ref| map_schedule_stop_point_quay.get(&ssp_ref))
+                    .and_then(|quay_ref| collections.stop_points.get_idx(&quay_ref))
+                {
+                    stop_point_idxs.push(stop_point_idx);
+                } else {
+                    return None;
+                }
+            }
+            if stop_point_idxs.is_empty() {
+                None
+            } else {
+                Some(stop_point_idxs)
+            }
+        })
+        .collect()
+}
+
+#[derive(Default, Debug)]
+struct TreeNode {
+    depth: usize,
+    is_leaf: bool,
+    childrens: HashMap<Idx<StopPoint>, TreeNode>,
+}
+type RoutingConstraintZones = TreeNode;
+
+// Transform a list of RoutingConstraintZone (which are basically an ordered list
+// of Stop Points) into a tree.
+impl From<Vec<RoutingConstraintZone>> for RoutingConstraintZones {
+    fn from(routing_constraint_zone_list: Vec<RoutingConstraintZone>) -> Self {
+        let mut routing_constraint_zones = RoutingConstraintZones::default();
+        for routing_constraint_zone in routing_constraint_zone_list {
+            let mut current_node = &mut routing_constraint_zones;
+            current_node.depth = 0;
+            for (depth, stop_point_idx) in routing_constraint_zone.iter().enumerate() {
+                current_node = current_node
+                    .childrens
+                    .entry(*stop_point_idx)
+                    .or_insert_with(TreeNode::default);
+                current_node.depth = depth + 1;
+                if depth == routing_constraint_zone.len() - 1 {
```

Comment:
you can do it like this :
   current_node.is_leaf = depth == routing_constraint_zone.len() - 1;

Reply:
But not exactly equivalent. Because if `current_node.depth` is already `true` and the `depth == routing_constraint_zone.len()-1` is false, your solution will erase it with `false` but mine will leave `true`.